### PR TITLE
Analytics: remove Redux indirection when getting superProps for a tracks event

### PR DESCRIPTION
**Pull request by jsnajdr**
When sending a tracks event, the analytics library calculates some additional props for the tracks event that are always the same. And for `calypso_page_view` events, adds some info about the selected site to the event if the recorded path has a site parameter (like `/posts/:site`). That feature is called "super props" and the additional props are provided by the `getAll` callback in `analytics/lib/super-props`.

This PR removes several levels of Redux indirection where `recordEvent` would dispatch a `ANALYTICS_SUPER_PROPS_UPDATE` action, a middleware would intercept that action, retrieve site info from Redux and call `analytics.setSelectedSite` and `analytics.setSiteCount`. Then, after the `dispatch` call, the analytics module has a new `_selectedSite` and `_siteCount` internal variable values.

After this patch, the `superProps` getter has the `reduxStore` available and can be called directly.

**Drive-by fixes:**
- simplify the library initialization: `analytics.initialize( ... )` works both in logged-out and logged-in mode, and `analytics.setDispatch` is not needed.
- during the initialization, the analytics library tries to detect if our stats scripts are blocked by a browser extension like uBlock and bumps a stat. (Done by @gravityrail in #17688) Before this patch, there was race condition where `checkForBlockedTracks` could be called before the
`loadScript` call is finished, and `_loadTracksError` wasn't set yet. Now we wait for the `loadScript` result. See 897fcfb, could be a standalone PR.

This PR is a very nice steps towards removing unneeded dependencies from analytics and making it a self-contained and reusable library.

**How to test:**
Enable tracks debugging in console:
```js
localStorage.debug = 'calypso:analytics:tracks'
```
and watch tracks events logged as you navigate Calypso. Check that `calypso_page_view` in site-specific section includes a `blog_id` prop, and it doesn't for other events. See also #23815.

### CODE REVIEW

**Comment by blowery** [comment]
This will need to take into account the changes in https://github.com/Automattic/wp-calypso/pull/36658 that also init analytics for the login entrypoint. Maybe we can find a better way to share that init code?

**Comment by jsnajdr** [comment]
> This will need to take into account the changes in #36658 that also init analytics for the login entrypoint.

I just updated the login entrypoint `analytics.initialize` call in cf96d98. That fixed the `site_count` super prop in the login events that are sent. That's because `site_count` is updated in the [`lib/middleware` handler for `ANALYTICS_SUPER_PROPS_UPDATE`](https://github.com/Automattic/wp-calypso/blob/831509ff42ef2d7a619e1981c9fc608282d197cf/client/state/lib/middleware.js#L216-L217) and that middleware is not present in the login entrypoint.

This patch removes the Redux indirection and `site_count` starts working again:

Before (broken, value always `0`):
Screenshot 2019-10-11 at 13 26 26

After (fixed, real value):
Screenshot 2019-10-11 at 12 54 20

@blowery @sgomes this is now ready for review again. It's one of the PRs that remove complexity from the analytics module and will eventually make it into a reasonable library 🙂

**Review by sgomes — APPROVED** [review]
The changes LGTM from a high-level point of view. This simplifies things a whole bunch, making the data flow a lot more sensible. Great work, @jsnajdr! 

From a tracking point of view, it appears to be working correctly, but it may be worth a look from someone that knows the details better than me.

File: client/lib/analytics/super-props.js [hunk 1]
```diff
@@ -1,38 +1,45 @@
...
-	},
+
+/**
+ * Internal dependencies
+ */
+import { shouldReportOmitBlogId } from 'lib/analytics/utils';
+import { getSelectedSite } from 'state/ui/selectors';
+import { getCurrentUserSiteCount } from 'state/current-user/selectors';
+
+const getSuperProps = reduxStore => eventProperties => {
```
[the remark this answers is not in the document]
Comment (jsnajdr) [reply client/lib/analytics/super-props.js]: In Java it would be `SuperPropsGetterFactory`, in JavaScript we have currying :wink:

**Pull request merged**